### ISSUE TITLE
Multiple inlet velocity vectors for outdoor airflow

### ISSUE THREAD

**Issue by saket411** [mostaphaRoudsari/butterfly#144]
I read through the old posts but could not find a proper solution to my problem.
I want to have the inlet wind profile as in this image -
![image](https://cloud.githubusercontent.com/assets/1322619/21669831/0a0dadb2-d34a-11e6-9c23-b2cffe5148b0.png)

Any hints how I can do this. Lets say have 10 different velocity vectors varying with height following the formula.

Thanks!!

**Comment by mimarilker** [comment]
Actually you can define a log. profile with an equation, however on wintunnel component there is an option named "landscape" and I think it does the same thing, if you check velocity probes you will see that there is already a wind speed profile through the wind tunnel domain. So I think thats enough, you may compare it with a logaritmic profile to be sure.

**Comment by TheodoreGalanos** [comment]
Hi @saket411 

There is no need to manually define a logarithmic boundary profile in OF. Butterfly is using the OF ABL function to assign automatically (to all outdoor cases) the ABL profile on the inlet. So as long as you are using the outdoor workflow you are assigning this profile to your inlet.

In OF the only thing you provide is typically the inlet velocity and reference height (for that velocity). OF takes care of velocity stratification.

Kind regards,
Theodore.

**Comment by saket411** [comment]
Yes I was just going to post that there is already a gradation by changing the landscape values. I just compared two of them.
![image](https://cloud.githubusercontent.com/assets/1322619/21672315/b7a4b36e-d35c-11e6-9d2d-8af94942f80b.png)
Looks like this should be enough.

**Comment by TheodoreGalanos** [comment]
Btw I would suggest the opposite configuration than the one I see in your screenshot. You need to have a small windward extension and large leeward extension of your WT. I would suggest 3x height for windward (distance from inlet to geometry) and 10-15x height for leeward (distance from geometry to outlet). You need to allow the outflow to stabilize.

**Comment by mimarilker** [comment]
Actually as Bert Blocken (has very very important studies on cfd) suggests as to the domain size, 5x height for windward (distance from inlet to geometry and x denotes for the tallest building height ) and 15x height for leeward (distance from geometry to outlet).Moreover 6x from ground plane, lastly 5x for both sides.

**Comment by TheodoreGalanos** [comment]
Hi @mimarilker Yes that is true, Blocken also references the same guidelines (Franke et al.) we used to assign these values. These guidelines however were based on a EU research project and were tailored to the EU region. The fault is mine really, I'm situated in the Tropics and I always think for the Tropics first. Wind velocities here are much much lower and we can get away with smaller values for the wind tunnel extensions.

@saket411 As @mimarilker suggests, if you are situated outside of the Tropics and the problem you are simulating has high average wind velocities then you should extend the boundaries even more. Even though usually results won't be impacted that much. It is actually a good experimentation for you to do if you have the time.

@mostaphaRoudsari I just realized our defaults are Tropics type defaults for sides and top. I don't think it is necessary to provide two default modes right? I think it's ok for users to experiment and learn by doing? What do you think.

Kind regards,
Theodore.

**Comment by TheodoreGalanos** [comment]
Hi @mimarilker Yes that is true, Blocken also references the same guidelines (Franke et al.) we used to assign these values. 

(I HIGHLY recommend these guidelines as reading material, a lot of useful information for case setup in there: https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=0ahUKEwiq-pCwy6rRAhXKqo8KHSdGAaEQFggeMAA&url=http%3A%2F%2Ftheairshed.com%2Fpdf%2FCOST%2520732%2520Best%2520Practice%2520Guideline%2520May%25202007.pdf&usg=AFQjCNEz8RAp5TujITU00MIx3iQ0TneCUQ&sig2=q5ahELyv4ph0wemcWQuUPg&cad=rja)

These guidelines however were based on a EU research project and were tailored to the EU region. The fault is mine really, I'm situated in the Tropics and I always think for the Tropics first. Wind velocities here are much much lower and we can get away with smaller values for the wind tunnel extensions.

@saket411 As @mimarilker suggests, if you are situated outside of the Tropics and the problem you are simulating has high average wind velocities then you should extend the boundaries even more. Even though usually results won't be impacted that much. It is actually a good experimentation for you to do if you have the time.

@mostaphaRoudsari I just realized our defaults are Tropics type defaults for sides and top. I don't think it is necessary to provide two default modes right? I think it's ok for users to experiment and learn by doing? What do you think.

Kind regards,
Theodore.

**Comment by saket411** [comment]
Actually the wireframe in the image above is the refinement box. The WT is as it is set in the example file (5,15,10).
But maybe I should take more test area for better visualization.
![image](https://cloud.githubusercontent.com/assets/1322619/21673476/0bb505ce-d364-11e6-8408-dc5bd142063f.png)

**Comment by saket411** [comment]
I am in Singapore. For urban level studies the codes say 'The surrounding buildings reside within 500 m distance stream wise and span wise from the edge of development of interest should be modelled explicitly'.

So they just need a clear 500m offset in x and y direction on all 4 sides. No mention of the height, but 5x should be enough.

**Comment by TheodoreGalanos** [comment]
You can also increase quality of results by multi-grading. I'm attaching a simple case with multiple grading, only on the Z axis. The settings roughly (I did it almost by hand) assign a 1m cell size on Z for the first 25 meters and a (total) cell expansion of 3 for the rest 50 meters.

P.S.: The case is also an example of wind comfort studies but you can ignore that and treat it as a single case.

Kind regards,
Theodore.
[170105_multigradient_example.zip](https://github.com/mostaphaRoudsari/butterfly/files/686908/170105_multigradient_example.zip)

**Comment by mimarilker** [comment]
@Theodore as you said multi-grading is of importance as to reduce calculation time. However it is better to say that this type of grading should be added for all axes to minimize calc. time. In detail, the mesh should expanse from the model geometry to the outer faces of domain, there may be a comp. doing this for user easily. Btw I always apply grading in Fluent and FloEFD.

On the other hand, to retain domain sizes in the guidelines very important, especially for outlet, it should be 15x, because OF accepts a zero value default inlet pressure, it means that at the outlet face there wont remain an effect from the model geometry, so we will see zero pressure again, make a research on mass conservation eqs. to have deeper knowledge on that.

**Comment by TheodoreGalanos** [comment]
@mimarilker Yes you are right this is an example of Z multi-grading. In normal, production cases, x,y, and z multi-grading should be applied (with care) since it really improves mesh quality without the need to increasing mesh cell count. I will try and add a true multi-grading example tomorrow. I think I can find the time to create it.

You are also correct on the second point, that is what I tried to awkwardly explain above (it was too early for physics lol) that the outflow boundary needs to be sufficiently away from the geometry so that there is a fully developed flow in the outflow. I have found that for most of my projects 15x max height was a bit of overkill. But again this is in the Tropics were CFD is a bit boring due to these low winds :)

**Comment by mimarilker** [comment]
waiting for multigrading component than, it will be amazing @Theodore, please let me know.

**Comment by TheodoreGalanos** [comment]
@mimarilker Yes that would be amazing. Unfortunately my biggest flaw is that I don't code so I couldn't really develop it. 

I have made a kind of algorithmic way to do it (i.e. a GH definition) but it's hacky. I think we could combine that and the amazing work here https://openfoamwiki.net/index.php/Scripts/blockMesh_grading_calculation to create a component.

Let me know if you want to assist on this, we can have a skype call on how to make it happen!

**Comment by saket411** [comment]
@**TheodoreGalanos** Did you do the GM in BF?

**Comment by TheodoreGalanos** [comment]
The case preparation yes. The model was a Rhino model so BF made it easy.

I almost always run my cases in Linux though, speed difference is quite substantial for that size of models (usually around 25-30 million cells). 

We did do a wind comfort and thermal comfort study in the same project though and there BF excelled. The first iteration of developing 36 cases for each wind direction took maybe a whole week (with errors and what not). The second iteration with BF took around 5 mins to create 36 folders, with a similar methodology as the file I attached above.

Kind regards,
Theodore.
